torch/lib/THCUNN/MSECriterion.cu

```diff
@@ -26,6 +26,19 @@ struct mse_functor
 };
 
 template <typename Dtype, typename Acctype>
+struct mse_functor_dtype
+{
+  mse_functor_dtype() {}
+
+  __host__ __device__ Dtype operator()(const Dtype &x, const Dtype &y) const
+  {
+    Acctype z = ScalarConvert<Dtype, Acctype>::to(x)-y;
```

Comment:
I think it would have been better to use the proper `add`/`sub` operations on `Dtype` instead of converting them to `Acctype` and then converting back. Even though this might not make a difference (because `Acctype` for float is float, and not double), I still think it would be cleaner.
Same thing for the following line.

Reply:
Actually, this might not work when `Dtype` is half. Maybe it might be better to use the THCNumerics here

Reply:
Good call, I didn't know about THCNumerics